components/bio-formats-tools/src/loci/formats/tools/XMLValidate.java

```diff
@@ -70,18 +115,7 @@ public static void main(String[] args) throws Exception {
     }
     else {
       // read from file(s)
-      for (int i=0; i<args.length; i++) {
-        if (args[i].toLowerCase().endsWith("tif") ||
-          args[i].toLowerCase().endsWith("tiff"))
-        {
-          String comment = new TiffParser(args[i]).getComment();
-          process(args[i], new BufferedReader(new StringReader(comment)));
-        }
-        else {
-          process(args[i], new BufferedReader(new InputStreamReader(
-            new FileInputStream(args[i]), Constants.ENCODING)));
-        }
-      }
+      validate(args);
```

Comment:
Probably a next step is to get the return of the `validate()` call and set `System.exit()` accordingly. can happen as a follow up.